lib/galaxy/actions/library.py

```diff
@@ -72,7 +74,7 @@ class LibraryActions:
     Mixin for controllers that provide library functionality.
     """
 
-    def _upload_dataset(self, trans, library_id, folder_id, replace_dataset=None, **kwd):
+    def _upload_dataset(self, trans, library_id: str, folder_id: str, replace_dataset: Optional[LibraryDataset] = None, **kwd):
```

Comment:
Looks like the `library_id` parameter is not really used in `_upload_dataset` and further down in `_create_folder` maybe we can safely delete them?

Reply:
Everything is green so I'm going to merge this - but I will follow up on this point! Thanks for the catch.

Reply:
It is in my deferred data branch (https://github.com/galaxyproject/galaxy/pull/12533/commits/f14c978e7e99f8512ba5b4e09ffc0ed551b21a09) - once things are green there again I'll PR it or PR it as part of other changes.